src/cpp/shared/combat/combat_ai.cpp

```diff
@@ -0,0 +1,294 @@
+#include "combat/combat.h"
+
+float heuristicModifierForDuration[] = {
+   0.0,
+   0.5,
+   0.65,
+   0.78,
+   0.85,
+   0.95,
+   1.03,
+   1.08,
+   1.12,
+   1.15,
+   1.18
+};
+
+float durationBenefit[] = {
+   0.0,
+   0.33,
+   0.55,
+   0.72,
+   0.85,
+   0.95,
+   1.03,
+   1.08,
+   1.12,
+   1.15,
+   1.18
+};
+
+void combatManager::DetermineEffectOfSpell(int spell, int *value, int *target) {
+  *value = 0;
+  AI_SPELL_TYPES aiSpellType;
+  int targetSide = 0;
+
+  switch(spell) {
+    case SPELL_MASS_SLOW:
+    case SPELL_MASS_CURSE:
+      aiSpellType = AI_MASS_DEBUFF_SPELL;
+      targetSide = 1 - this->currentActionSide;
+      break;
+    case SPELL_SLOW:
+    case SPELL_CURSE:
+      aiSpellType = AI_TARGET_DAMAGE_OR_DEBUFF_SPELL;
+      targetSide = 1 - this->currentActionSide;
+    default:
+      this->DetermineEffectOfSpell_orig(spell, value, target);
+      return;
+  }
+
+  int targetHex = 0;
+  int hex = 1;
+
+  if(aiSpellType == AI_TARGET_BUFF_SPELL
+    || aiSpellType == AI_TARGET_DAMAGE_OR_DEBUFF_SPELL
+    || aiSpellType == AI_MASS_DEBUFF_SPELL
+    || aiSpellType == AI_MASS_BUFF_SPELL
```

Comment:
As written, only AI_MASS_DEBUFF_SPELL will reach this line. What's with the rest of this?